test/unit/test_callables.py

```diff
@@ -0,0 +1,121 @@
+# This file is part of PyOP2
+#
+# PyOP2 is Copyright (c) 2012-2014, Imperial College London and
+# others. Please see the AUTHORS file in the main source directory for
+# a full list of copyright holders.  All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#     * The name of Imperial College London or that of other
+#       contributors may not be used to endorse or promote products
+#       derived from this software without specific prior written
+#       permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTERS
+# ''AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import pytest
+import loopy
+from pyop2.codegen.rep2loopy import inv_fn_lookup, solve_fn_lookup
+import numpy as np
+from pyop2 import op2
+
+
+@pytest.fixture
+def s():
+    return op2.Set(1)
+
+
+@pytest.fixture
+def zero_mat(s):
+    return op2.Dat(s ** (2, 2), [[0.0, 0.0], [0.0, 0.0]])
+
+
+@pytest.fixture
+def inv_mat(s):
+    return op2.Dat(s ** (2, 2), [[1.0, 2.0], [3.0, 4.0]])
+
+
+@pytest.fixture
+def zero_vec(s):
+    return op2.Dat(s ** (2, 1), [0.0, 0.0])
+
+
+@pytest.fixture
+def solve_mat(s):
+    d = op2.Dat(s ** (2, 2), [[2.0, 3.0], [-1.0, 2.0]])
```

Comment:
The "solve" test fails. The result gives a solution for a system matrix that is the transpose of this matrix. Did I do something wrong on PyOP level in the setup of the test case?

Reply:
Resolved. I replaced gesv with a getrs call such that we can specify to transpose. It was a row major vs column major issue.

Reply:
Oh yeah, fortran.